packages/core/src/blocks/defaultBlockTypeGuards.ts

```diff
@@ -2,53 +2,179 @@ import { CellSelection } from "prosemirror-tables";
+    } else {
+      if (
+        editor.schema.blockSpecs[blockType].config.propSchema[propName]
+          .default !== propSpec.default
+      ) {
+        return false;
+      }
+
+      if (
+        editor.schema.blockSpecs[blockType].config.propSchema[propName]
+          .default === undefined &&
+        propSpec.default === undefined
+      ) {
+        if (
+          editor.schema.blockSpecs[blockType].config.propSchema[propName]
+            .type !== propSpec.type
+        ) {
+          return false;
+        }
+      }
+
+      if (
+        typeof editor.schema.blockSpecs[blockType].config.propSchema[propName]
+          .values !== typeof propSpec.values
+      ) {
+        return false;
+      }
+
+      if (
+        typeof editor.schema.blockSpecs[blockType].config.propSchema[propName]
+          .values === "object" &&
+        typeof propSpec.values === "object"
+      ) {
+        if (
+          editor.schema.blockSpecs[blockType].config.propSchema[propName].values
+            .length !== propSpec.values.length
+        ) {
+          return false;
+        }
+
+        for (
+          let i = 0;
+          i <
+          editor.schema.blockSpecs[blockType].config.propSchema[propName].values
+            .length;
+          i++
+        ) {
+          if (
+            editor.schema.blockSpecs[blockType].config.propSchema[propName]
+              .values[i] !== propSpec.values[i]
+          ) {
+            return false;
+          }
+        }
+      }
+    }
+  }
+
+  return true;
 }
 
-export function checkBlockTypeInSchema<
-  BlockType extends string,
-  Config extends BlockConfig,
+export function blockHasType<
+  BType extends string,
+  Props extends
+    | PropSchema
+    | Record<string, "boolean" | "number" | "string">
+    | undefined = undefined,
 >(
-  blockType: BlockType,
-  blockConfig: Config,
+  block: Block<any, any, any>,
   editor: BlockNoteEditor<any, any, any>,
-): editor is BlockNoteEditor<{ [T in BlockType]: Config }, any, any> {
+  blockType: BType,
+  props?: Props,
+): block is Block<
+  {
+    [BT in BType]: Props extends PropSchema
+      ? BlockConfig<BT, Props>
+      : Props extends Record<string, "boolean" | "number" | "string">
+        ? BlockConfig<
+            BT,
+            {
+              [PN in keyof Props]: PropSpec<
+                Props[PN] extends "boolean"
+                  ? boolean
+                  : Props[PN] extends "number"
+                    ? number
+                    : Props[PN] extends "string"
+                      ? string
+                      : never
+              >;
+            }
+          >
+        : BlockConfig<BT, PropSchema>;
+  },
+  any,
+  any
+> {
   return (
-    blockType in editor.schema.blockSchema &&
-    editor.schema.blockSchema[blockType] === blockConfig
+    editorHasBlockWithType(editor, blockType, props) && block.type === blockType
   );
 }
 
+// TODO: Only used once in the emoji picker - is it even needed? If so, should
```

Comment:
@nperez0111, any thoughts on this btw?

Reply:
I would move it to just the emoji picker. If nothing else needs it right now, let's keep it simple